### PR TITLE
10382 golden ratio limit

### PR DESCRIPTION
**Pull request by sagar-bharadwaj-ks**
#10382

### CODE REVIEW

**Comment by SagarB-97** [comment]
Some consequences of changes : 

```python
>>> limit(fibonacci(n+1)/fibonacci(n),n,oo)
GoldenRatio
>>> limit(fibonacci(n)/fibonacci(n-1),n,oo)
GoldenRatio
>>> limit(fibonacci(n+2)/fibonacci(n),n,oo)
GoldenRatio**2
>>> limit(fibonacci(n+i)/fibonacci(n),n,oo)
GoldenRatio**i
```

**Comment by SagarB-97** [comment]
@asmeurer @jksuom 
The above fix works for various definitions of `n` including 
```python
n = symbols('n')
```
and
```python
n = Symbol('n' , integer=True)
```

However, a `NotImplementedError()` is raised if `n` is defined as : 
```python
n = Symbol('n' , positive=True, integer=True)
```
or 
```python
n = Symbol('n' , positive = True)
```

I think this might be a bug in the `doit()` function. The expression `fibonacci(n+1)/fibonacci(n)` only gets converted to an expression containing `GoldenRatio` raised to various `n` powers.

File: sympy/series/limits.py [hunk 1]
```diff
@@ -144,6 +146,12 @@ def doit(self, **hints):
         if z0.is_positive:
             e = e.rewrite([factorial, RisingFactorial], gamma)
 
+        # limits of fibonacci series when the variable approaches infinity
+        # can be conviniently calculated if fibonacci is rewritten in closed
+        # form in terms of Golden Ratio
+        if z0.is_infinite:
+            e = e.rewrite(fibonacci,GoldenRatio)
```
Comment (jksuom) [inline sympy/series/limits.py]: Please add a space after comma.

This rewrite line could probably be placed after the factor_terms line below when `z0` has already been checked. (The comment may not be needed either.)

File: sympy/series/tests/test_limits.py [hunk 2]
```diff
@@ -105,6 +107,10 @@ def eval(cls, arg):
 def test_issue_3885():
     assert limit(x*y + x*z, z, 2) == x*y + 2*x
 
+def test_issue_10382():
+    n = Symbol('n', integer=True)
+    assert limit(fibonacci(n+1)/fibonacci(n),n,oo) == S.GoldenRatio
```
Comment (jksuom) [inline sympy/series/tests/test_limits.py]: Spaces after commas here as well.

**Comment by SagarB-97** [comment]
@jksuom 
Any more changes required?

**Comment by jksuom** [comment]
Thanks! I think this can be merged now as it is. The existence of the limit depends on the rather fragile `ok` test since `gruntz` cannot handle it. If it is later improved, this rewrite may be replaced by the 'tractable' rewrite in `gruntz`.

**Comment by SagarB-97** [comment]
@jksuom 
Can #10382 be closed now?

**Comment by jksuom** [comment]
Yes. It does happen automatically when a comment contains 'fixes #xxxxx'.

**Pull request merged**